dev-packages/cloudflare-integration-tests/runner.ts

```diff
@@ -0,0 +1,237 @@
+
+      let envelopeCount = 0;
+      const { resolve: setWorkerPort, promise: workerPortPromise } = deferredPromise<number>();
+      let child: ReturnType<typeof spawn> | undefined;
+
+      /** Called after each expect callback to check if we're complete */
+      function expectCallbackCalled(): void {
+        envelopeCount++;
+        if (envelopeCount === expectedEnvelopeCount) {
+          resolve();
+        }
+      }
+
+      function newEnvelope(envelope: Envelope): void {
+        if (process.env.DEBUG) log('newEnvelope', inspect(envelope, false, null, true));
+
+        const envelopeItemType = envelope[1][0][0].type;
+
+        if (ignored.has(envelopeItemType)) {
+          return;
+        }
+
+        const expected = expectedEnvelopes.shift();
+
+        // Catch any error or failed assertions and pass them to done to end the test quickly
+        try {
+          if (!expected) {
+            return;
+          }
+
+          if (typeof expected === 'function') {
+            expected(envelope);
+          } else {
+            expect(envelope).toEqual(expected);
+          }
+          expectCallbackCalled();
+        } catch (e) {
+          reject(e);
+        }
+      }
+
+      createBasicSentryServer(newEnvelope)
+        .then(([mockServerPort, mockServerClose]) => {
+          if (mockServerClose) {
+            CLEANUP_STEPS.add(() => {
+              mockServerClose();
+            });
+          }
+
+          if (process.env.DEBUG) log('Starting scenario', testPath);
+
+          const stdio: ('inherit' | 'ipc' | 'ignore')[] = process.env.DEBUG
+            ? ['inherit', 'inherit', 'inherit', 'ipc']
+            : ['ignore', 'ignore', 'ignore', 'ipc'];
+
+          child = spawn(
+            'wrangler',
+            [
+              'dev',
+              '--config',
+              join(testPath, 'wrangler.jsonc'),
+              '--show-interactive-dev-session',
+              'false',
+              '--var',
+              `SENTRY_DSN:http://public@localhost:${mockServerPort}/1337`,
+            ],
+            { stdio },
+          );
+
+          CLEANUP_STEPS.add(() => {
+            child?.kill();
+          });
+
+          child.on('error', e => {
+            // eslint-disable-next-line no-console
+            console.error('Error starting child process:', e);
+            reject(e);
+          });
+
+          child.on('message', (message: string) => {
+            const msg = JSON.parse(message) as { event: string; port?: number };
+            if (msg.event === 'DEV_SERVER_READY' && typeof msg.port === 'number') {
+              setWorkerPort(msg.port);
+              if (process.env.DEBUG) log('worker ready on port', msg.port);
+            }
+          });
+        })
+        .catch(e => reject(e));
+
+      return {
+        completed: async function (): Promise<void> {
+          return isComplete;
+        },
+        makeRequest: async function <T>(
```

Comment:
BTW, maybe it could be possible to have a WS method also?

Reply:
Yes. I guess we'll add more options when we add tests that need it!